streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -249,10 +254,15 @@ public void init(final ProcessorContext context,
         // open the DB dir
         metricsRecorder.init(getMetricsImpl(context), context.taskId());
         openDB(context.appConfigs(), context.stateDir());
+        this.position = StoreQueryUtils.readPositionFromCheckpoint(positionCheckpoint);
 
         // value getter should always read directly from rocksDB
         // since it is only for values that are already flushed
-        context.register(root, (RecordBatchingStateRestoreCallback) this::restoreBatch);
+        context.register(root, (RecordBatchingStateRestoreCallback) this::restoreBatch, this::checkpoint);
```

Comment:
Sorry; one more formatting correction. This applies in several places in this PR:

> If a method call is longer than 120 characters, switch to a single parameter per line formatting (instead of just breaking it into two lines only).

https://kafka.apache.org/coding-guide.html